### ISSUE TITLE
Will the JobDispatcher ever be executed?

### ISSUE THREAD

**Issue by Amosss** [firebase/firebase-jobdispatcher-android#129]
I understand that in order to create a JobDispatcher, I need to write the following in onCreate of MainActivity:

FirebaseJobDispatcher dispatcher = new FirebaseJobDispatcher(new GooglePlayDriver(this));    
Job myJob = dispatcher.newJobBuilder()
    .setService(theService.class)
    .setTag("thetag")
    **.setRecurring(true)**
    **.setLifetime(Lifetime.FOREVER)**
    **.setTrigger(Trigger.executionWindow(604800, 604860))**
    **.setReplaceCurrent(true)**
    .setRetryStrategy(RetryStrategy.DEFAULT_LINEAR)
    .setConstraints(Constraint.ON_ANY_NETWORK)
    .build();

In short, I want a process to run once a week, to last forever, to be recurrent and replace the current one if/when created again. Won't the above means that every time MainActivity is created, it re-creates (overwrite) the previous job? This way it will never be executed because the "timer" of 1 week will start from the beginning. Am I missing something?

**Comment by unEgor** [comment]
Hi Amosss,

I'm not sure that I understand your question correctly.
Why do you need to replace current job?

If you set replace current to true it means that your job timer starts from 0.

**Comment by Amosss** [comment]
Thanks.

So if I understand your answer, I should change it to .setReplaceCurrent(false) so my previous creation will remain intact?

What if at a later version of my app, I will change one of the definitions and want to overwrite the old one? How will I be able to do so?

**Comment by unEgor** [comment]
When you package was updated all jobs should be gone.
You could listen for https://developer.android.com/reference/android/content/Intent.html#ACTION_MY_PACKAGE_REPLACED and reschedule your job if needed.

**Comment by Amosss** [comment]
So whenever I update the app on a mobile phone, all the JobDispatchers it had are deleted and once MainActivity will run again, it will re-create the job with it? right?

I defined it as .setLifetime(Lifetime.FOREVER) so I thought it really lasts forever and it's up to me to control its existence.

Thanks!

**Comment by FabianTerhorst** [comment]
You can define an BroadcastReceiver that executes when the given action happens for the app and reschedule the jobs inside.

**Comment by samtstern** [comment]
In April 2019 we announced that Firebase Job Dispatcher would be deprecated today, April 7th 2020.  For this reason we are going to close all open issues and archive the repository.  We will also stop supporting FJD for new app installations (such as those targeting sdk versions greater than Android Q).  Existing apps that rely on FJD should continue to function as usual.

While we will no longer work on FJD, we encourage you to [migrate to Work Manager](https://developer.android.com/topic/libraries/architecture/workmanager/migrating-fb) which is a superior tool and fully supported by the Android team.

Thank you for taking the time to try the library and provide this feedback.  We sincerely appreciate your effort and we apologize for not addressing it in time.
